src/frontends/lean/widget.cpp

```diff
@@ -315,6 +341,56 @@ std::vector<vdom> render_html_list(vm_obj const & htmls, std::vector<component_i
     return elements;
 }
 
+json get_effect(vm_obj const & o_effects) {
+    json result = json::array();
+    list<vm_obj> effects = to_list(o_effects);
+    for (auto e : effects) {
+        switch (cidx(e)) {
+            case effect_idx::insert_text: {
+                result.push_back({
+                    {"kind", "insert_text"},
+                    {"text", to_string(cfield(e, 0))}
+                });
+                break;
+            } case effect_idx::reveal_position: {
+                auto pos = to_pos_info(cfield(e, 1));
```

Comment:
I get a vm_check failed here because `cfield(e,1)` is not a vm_composite for the `widget_effects` test. :/